model.py

```diff
@@ -1,103 +1,1394 @@
+            return -np.inf
+
+        ##############################################
+        # ~~~~~ Is the BS scale physically OK? ~~~~~ #
+        ##############################################
+        # We're gonna check to see if the BS scle is #
+        # in the range rwd/3 < (BS scale) < rwd*3.   #
+        # If it isn't, then it's either too          #
+        # concentrated to make sense, or so large    #
+        # that our approximation of a smooth disc is #
+        # no longer a good idea.                     #
+        #                                            #
+        ##############################################
+
+        # Get the WD radius.
+        rwd = ancestor_param_dict['rwd'].currVal
+
+        # Enforce the BS scale being within these limits
+        rmax = rwd * 3.
+        rmin = rwd / 3.
+
+        scale = ancestor_param_dict['scale'].currVal
+
+        if scale > rmax or scale < rmin:
+            if verbose:
+                print("Leaf {} has a BS scale that lies outside valid range!".format(self.name))
+                print("Rwd: {:.3f}".format(rwd))
+                print("Scale: {:.3f}".format(scale))
+                print("Range: {:.3f} - {:.3f}".format(rmin, rmax))
+
+            return -np.inf
+
+        ##############################################
+        # ~~~~~ Does the stream miss the disc? ~~~~~ #
+        ##############################################
+
+        slope = 80.0
+        try:
+            # q, rdisc_a were previously retrieved
+            az = ancestor_param_dict['az'].currVal
+
+            # If the stream does not intersect the disc, this throws an error
+            x, y, vx, vy = roche.bspot(q, rdisc_a)
+
+            # Find the tangent to the disc
+            alpha = np.degrees(np.arctan2(y, x))
+
+            # If alpha is negative, the BS lags the disc.
+            # However, the angle has to be less than 90 still!
+            if alpha < 0:
+                alpha = 90 - alpha
+
+            # Disc tangent
+            tangent = alpha + 90
+
+            # Calculate the min and max azimuthes, using the tangent and slope
+            minaz = max(0, tangent-slope)
+            maxaz = min(178, tangent+slope)
+
+            if az < minaz or az > maxaz:
+                return -np.inf
+
+        except:
+            if verbose:
+                print("The mass stream of leaf {} does not intersect the disc!".format(self.name))
+            return -np.inf
+
+        # If we pass all that, then calculate the ln_prior normally
+        lnp = super().ln_prior(verbose=verbose, *args, **kwargs)
+
+        return lnp
+
+    def plot_data(self, save=False, figsize=(11., 8.), fname=None, save_dir='.'):
+        '''Create a plot of the eclipse's data.
+
+        If save is True, a copy of the figure is saved.
+
+        If fname is defined, save the figure with that filename. Otherwise,
+        infer one from the data filename
+        '''
+
+        cv_par_name_list = [
```

Comment:
Would be neater to make this a property of the ```Eclipse```, since you use this code several times